### PR TITLE
use magicdns for harnesses using docker

### PR DESCRIPTION
**Pull request by joshrwolf**
hardcode the equivalent of an: `--add-host=host.docker.internal:host-gateway` to containers that `imagetest` spins up.

this means we can pretty much ~always assume we can access containers the same way we would from the host within containers without needing all the network configuration that such a pita for the user to configure. in layman's terms, this means no more need for specifying `networks` when dealing with local registries 

as far as I can tell, there's no downside to this on daemons > 20.04, which is over a year old at this point.

https://docs.docker.com/desktop/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host

### CODE REVIEW

**Review by priyawadhwa — APPROVED** [review]

**Comment by shimmerjs** [comment]
Don't we want to resist leaking into the host as much as possible? Currently reading up on the link you sent to better understand what exactly `host.docker.internal:host-gateway` is
